### ISSUE TITLE
Compilation with mingw64 6.3.0 fails due to constexpr in WrappedMutex constructor

### ISSUE THREAD

**Issue by fchucholowski** [protocolbuffers/protobuf#15331]
**What version of protobuf and what language are you using?**
Version: main/v3.21.12
Language: C++

**What operating system (Linux, Windows, ...) and version?**
Windows 11

**What runtime / compiler are you using (e.g., python version or gcc version)**
mingw64 6.3.0

**What did you do?**
I tried to compile the conan package `protobuf/3.21.12` with the mingw64 compiler.

**What did you expect to see**
Compilation works without errors.

**What did you see instead?**
I received following error
```
C:\tools\dyna4-build-mingw-w64\bin\x86_64-w64-mingw32-g++.exe --sysroot=C:/tools/build-mingw-w64/x86_64-w64-mingw32 -DGOOGLE_PROTOBUF_CMAKE_BUILD -DHAVE_ZLIB -ID:/dev/.conan/data/zlib/1.3/_/_/package/a2678573f9645761cbc6be1f8e53431ce61b150b/include -ID:/dev/.short/fa78a1/1/build/Release -ID:/dev/.short/fa78a1/1/src/src -m64 -fpermissive -O3 -DNDEBUG -std=c++11 -MD -MT CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/arena.cc.obj -MF CMakeFiles\libprotobuf-lite.dir\src\google\protobuf\arena.cc.obj.d -o CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/arena.cc.obj -c D:/dev/.short/fa78a1/1/src/src/google/protobuf/arena.cc
In file included from D:/dev/.short/fa78a1/1/src/src/google/protobuf/arena.cc:44:0:
D:/dev/.short/fa78a1/1/src/src/google/protobuf/stubs/mutex.h: In constructor 'constexpr google::protobuf::internal::WrappedMutex::WrappedMutex()':
D:/dev/.short/fa78a1/1/src/src/google/protobuf/stubs/mutex.h:124:29: error: temporary of non-literal type 'google::protobuf::internal::CallOnceInitializedMutex<std::mutex>' in a constant expression
   constexpr WrappedMutex() {}
                             ^
D:/dev/.short/fa78a1/1/src/src/google/protobuf/stubs/mutex.h:98:7: note: 'google::protobuf::internal::CallOnceInitializedMutex<std::mutex>' is not literal because:
 class CallOnceInitializedMutex {
       ^~~~~~~~~~~~~~~~~~~~~~~~
D:/dev/.short/fa78a1/1/src/src/google/protobuf/stubs/mutex.h:98:7: note:   'google::protobuf::internal::CallOnceInitializedMutex<std::mutex>' has a non-trivial destructor
```

**Anything else we should know about your project / environment**
There seems to be a special case for `qnx` compiler already in line 122. When I append a ` || defined(__MINGW32__)` in line 121, compilation works without problems.

**Comment by acozzette** [comment]
Could you try with 3.25 or main? We are unlikely to do any more patch releases on 3.21 at this point, and the problem should disappear in newer versions as we have since deleted mutex.h in favor of Abseil.

**Comment by fchucholowski** [comment]
Thank you for the quick reply. I think that this problem will be solved then. Unfortunately there is no conan package available for 3.25 yet. So for now I'm going to stick to 3.21 and patch mutex.h, which I already implemented.
I'll try to check 3.25 though soon. If there are any other problems, I'll file a new issue.

**Comment by acozzette** [comment]
Sounds good. We actually don't own the Conan package, so unfortunately we can't say when a newer version will be available. If you need to use Conan then sticking with a patch is probably the way to go for now.
